src/io/file_descriptor.cr

```diff
@@ -147,6 +147,7 @@ class IO::FileDescriptor < IO
   end
 
   def reopen(other : IO::FileDescriptor)
+    return if self.fd == other.fd
```

Comment:
should this return self?
will this introduce a `Nil` into the types that can be returned?

Reply:
Thanks for spotting! (this method should probably return nil but thats for another PR.